eland/tests/dataframe/test_metrics_pytest.py

```diff
@@ -250,3 +229,373 @@ def test_metric_agg_keep_dtypes(self):
             "Cancelled": {"max": True, "median": False, "min": False},
             "dayOfWeek": {"max": 0, "median": 0, "min": 0},
         }
+
+    def test_flights_numeric_only(self):
+        filter_data = [
+            "AvgTicketPrice",
+            "Cancelled",
+            "dayOfWeek",
+            "timestamp",
+            "DestCountry",
+        ]
+        # All Aggregations Data Check
+        ed_flights = self.ed_flights().filter(filter_data)
+        pd_flights = self.pd_flights().filter(filter_data)
+        # agg => numeric_only True returns float64 values
+        # We compare it with individual non-agg functions of pandas with numeric_only=True
+        # not checking mad because it returns nan value for booleans.
+        filtered_aggs = self.funcs + self.extended_funcs
+        filtered_aggs.remove("mad")
+        agg_data = ed_flights.agg(filtered_aggs, numeric_only=True).transpose()
+        for agg in filtered_aggs:
+            assert_series_equal(
+                agg_data[agg].rename(None),
+                getattr(pd_flights, agg)(
+                    **({"numeric_only": True} if agg != "mad" else {})
+                ),
+                check_exact=False,
+                rtol=True,
+            )
+
+    # Mean
+    @pytest.mark.parametrize("numeric_only", [True, False, None])
+    def test_mean_numeric_only(self, numeric_only):
```

Comment:
These tests are quite long and unwieldly, is there any way we can test the output (shape, dtype, etc) instead of testing values exhaustively?

Reply:
Okay! I just added the values and individual values dtype check, so that we are maintaining dtype for that individual column. 
Should I remove those and modify the tests so to check
- dtype of returning frame or series.
- shape ( the no. of rows and column)